xbmc/filesystem/DllLibCurl.cpp

```diff
@@ -59,38 +60,124 @@ unsigned long ssl_thread_id(void)
 
 #endif // HAVE_OPENSSL
 
-using namespace XCURL;
+namespace XCURL
+{
+#define CURL CURL_HANDLE
+#include <curl/curl.h>
+#undef CURL
 
-/* okey this is damn ugly. our dll loader doesn't allow for postload, preunload functions */
-static long g_curlReferences = 0;
-#if(0)
-static unsigned int g_curlTimeout = 0;
-#endif
+CURLcode DllLibCurl::global_init(long flags)
+{
+  return curl_global_init(flags);
+}
 
-bool DllLibCurlGlobal::Load()
+void DllLibCurl::global_cleanup()
 {
-  CSingleLock lock(m_critSection);
-  if(g_curlReferences > 0)
-  {
-    g_curlReferences++;
-    return true;
-  }
+  curl_global_cleanup();
+}
 
-  /* we handle this ourself */
-  DllDynamic::EnableDelayedUnload(false);
-  if (!DllDynamic::Load())
-    return false;
+CURL_HANDLE * DllLibCurl::easy_init()
```

Comment:
do we really need this wrapper class now ?

Reply:
At this stage not really. I was thinking of making the class into a more designed wrapper that keeps track of easy/multi handles and encapsulates everything curl in one place.
I also want to get rid of the state in curl global but it requires more work so it'll have to wait for a separate pr.

Reply:
okay, if it's a todo..